app/geo/gui/src/ts/twxplore/gui/geo/attributeStrategies/attributeStrategies.ts

```diff
@@ -1,82 +1,156 @@
 // In an OOP Language -
+
+import { FeatureAttributeName } from "../states/map/FeatureAttributeName";
+
 // TypeScript
-interface FeatureAttributeFunctions {
+interface FeatureAttributeTypes {
     isNumeric() : boolean
     isString(): boolean
+    getFilterType(): string
+    ignore(): boolean
+    attributeName: FeatureAttributeName
 }
 
-enum Attributes {
-    LABEL = "label",
-    TYPE = "type",
-    URI = "uri",
-    FREQUENCY = "frequency",
-    TIMESTAMP = "timestamp",
-    LOCALITY = "locality",
-    POSTALCODE = "postalcode",
-    REGIONS =  "regions",
-    TRANSMISSIONPOWER = "transmissionPower",
-    GEOMETRY = "geometry",
-}
 
-export class FeatureAttribute implements FeatureAttributeFunctions{
-    private attributeName: string 
-    private subclass: any
-    constructor(attributeName: string) {
+ export abstract class FeatureAttribute implements FeatureAttributeTypes{
+    public attributeName: FeatureAttributeName 
+    constructor(attributeName: FeatureAttributeName) {
         this.attributeName = attributeName
-        this.subclass = null
-        switch (this.attributeName) {
-            case (Attributes.TIMESTAMP): {
-                this.subclass = FeatureAttributeTimestamp.getInstance()
+    }
+    static AttributeTypeOf(attributeName: FeatureAttributeName): FeatureAttribute {
```

Comment:
camelCase method names.

Reply:
Needs a better name anyway.
I would suggest byName(name: FeatureAttributeName) or valueOf(name: FeatureAttributeName). The latter is the Java name.